packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1748,36 +1779,53 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
 
     assert(_debugCheckLocalizations(appLocale));
 
-    return RootRestorationScope(
-      restorationId: widget.restorationScopeId,
-      child: SharedAppData(
-        child: Shortcuts(
-          debugLabel: '<Default WidgetsApp Shortcuts>',
-          shortcuts: widget.shortcuts ?? WidgetsApp.defaultShortcuts,
-          // DefaultTextEditingShortcuts is nested inside Shortcuts so that it can
-          // fall through to the defaultShortcuts.
-          child: DefaultTextEditingShortcuts(
-            child: Actions(
-              actions: widget.actions ?? <Type, Action<Intent>>{
-                ...WidgetsApp.defaultActions,
-                ScrollIntent: Action<ScrollIntent>.overridable(context: context, defaultAction: ScrollAction()),
-              },
-              child: FocusTraversalGroup(
-                policy: ReadingOrderTraversalPolicy(),
-                child: TapRegionSurface(
-                  child: ShortcutRegistrar(
-                    child: Localizations(
-                      locale: appLocale,
-                      delegates: _localizationsDelegates.toList(),
-                      child: title,
-                    ),
-                  ),
+    Widget child = Shortcuts(
+      debugLabel: '<Default WidgetsApp Shortcuts>',
+      shortcuts: widget.shortcuts ?? WidgetsApp.defaultShortcuts,
+      // DefaultTextEditingShortcuts is nested inside Shortcuts so that it can
+      // fall through to the defaultShortcuts.
+      child: DefaultTextEditingShortcuts(
+        child: Actions(
+          actions: widget.actions ?? <Type, Action<Intent>>{
+            ...WidgetsApp.defaultActions,
+            ScrollIntent: Action<ScrollIntent>.overridable(context: context, defaultAction: ScrollAction()),
+          },
+          child: FocusTraversalGroup(
+            policy: ReadingOrderTraversalPolicy(),
+            child: TapRegionSurface(
+              child: ShortcutRegistrar(
+                child: Localizations(
+                  locale: appLocale,
+                  delegates: _localizationsDelegates.toList(),
+                  child: title,
                 ),
               ),
             ),
           ),
         ),
       ),
     );
+
+    if (widget.onNavigationNotification != null) {
+      child = NotificationListener<NavigationNotification>(
+        onNotification: widget.onNavigationNotification == null ? null : (NavigationNotification notification) {
+          // By default, don't do anything with navigation notifications if
+          // there is no engine attached.
+          if (widget.onNavigationNotification == WidgetsApp.defaultOnNavigationNotification
```

Comment:
This is weird can we work around this by making this non static? or a static method that takes in a widget state or changenotifier  and returns a function ?

Reply:
It turns out I do need this fix.  I've cleaned it up and made it non-static in this commit: https://github.com/flutter/flutter/pull/132249/commits/b33289698f1504b10d91803a71fa1052cf2fd0e3